vmdk_plugin/refcnt.go

```diff
@@ -150,18 +169,24 @@ func (r refCountsMap) Init(d *vmdkDriver) {
 		return
 	}
 
-	log.Infof("Discovered %d volumes in use.", len(r))
-	for name, cnt := range r {
+	// RLocks the refCountsMap
+	r.mtx.RLock()
+	defer r.mtx.RUnlock()
+	log.Infof("Discovered %d volumes in use.", len(r.refMap))
+	for name, cnt := range r.refMap {
 		log.Infof("Volume name=%s count=%d mounted=%t device='%s'",
 			name, cnt.count, cnt.mounted, cnt.dev)
 	}
 }
 
 // Returns ref count for the volume.
 // If volume is not referred (not in the map), return 0
-// NOTE: this assumes the caller holds the lock if we run concurrently
 func (r refCountsMap) getCount(vol string) uint {
-	rc := r[vol]
+	// RLocks the refCountsMap
+	r.mtx.RLock()
```

Comment:
can't another thread change the value right after getCount returns and unlocks, but before the value is used for comparison plugin.go:Remove() ?